### PR TITLE
utils.software_manager: introduce extract_from_package()

### LINKED ISSUE #3549: utils/software_management: Add extract_from_deb and extract_from_rpm operations
### Context ### 

The QEMU acceptance tests [1] provides some facilities for extracting files from either RPM and Deb packages. Those files are usually compiled Linux kernels that are used on 'boot an VM' kind of tests.

Those facilities are implemented in two methods in the `boot_linux_console.py` test module:
```
extract_from_deb(self, deb, path)

extract_from_rpm(self, rpm, path)
```
 

### Problem ###

Maintain that kind of facility isn't really on the scope of QEMU acceptance tests project, they would be better kept in Avocado. Also it could benefit Avocado users given that those utilities are pretty generic.

### Proposal ###

1) Move `extract_from_[rpm|deb]` to Avocado's `utils/software_management.py` module. 

The `utils/software_management.py` provides the `RpmBackend` and `DpkgBackend` implementations for, respectively, `rpm` and `dpkg` commands. Those backend seems to be good places to put those methods in. Or, alternatively, there could be created classes that represent an *rpm* or *deb* file and provides operations.   
 
2) Improve `extract_from_[rpm|deb]` method

Both methods extract a single file (passed via `path` parameter) but it could be useful to allow extract more than one file at once. Or the user could not even pass `path`, in this case it simply decompress/extract the whole package.

It needs to properly raise an exception or provides other mean to notify the user that the operation failed. 

### References ###

[1] https://github.com/qemu/qemu/tree/master/tests/acceptance

### PR DESCRIPTION
**Pull request by beraldoleal**
This method will extract the package content into a destination
directory. It isn't using the self discovery method, because is distro based 
and here we should try to extract packages even when in a different distro.
Fixes #3549.

Signed-off-by: Beraldo Leal <bleal@redhat.com>

### CODE REVIEW

**Comment by clebergnu** [comment]
@beraldoleal thanks for this! I haven't got to the review yet, but please consider adding some tests.  One comment is that we have a number of sample files under `selftests/.data`, where you could also add a minimal `.deb` and `.rpm` file.

**Review by ana — APPROVED** [review]
LGTM, thanks!
I agree adding some tests is a good idea.

**Comment by beraldoleal** [comment]
@clebergnu added some basic tests.

File: selftests/unit/utils/test_software_manager.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+import shutil
```
Comment (willianrampazzo) [inline selftests/unit/utils/test_software_manager.py:2]: As this test is depending on external artifacts, I think it fits better under `functional`.

**Comment by beraldoleal** [comment]
Closing this in favor of a v2.

**Pull request closed without merging**